Desktop/pm 18769/migrate vault filters

lxiong-livefront opened the pull request:
## 🎟️ Tracking

[JIRA story](https://bitwarden.atlassian.net/browse/PM-18769?atlOrigin=eyJpIjoiNzBhZDQ3NTU4Mjk3NDZkZjliM2I3YTM3OTgwZWM0ZTMiLCJwIjoiaiJ9)

## 📔 Objective

Migrated vault filters to new v3 vault's navigation
- Decoupled existing vault filtering from vault component by using routed params with `routed-vault-filter-bridge`
- Converted vault filters to standalone components
- Removed extending filter Base Components from deprecated `/libs/angular` library and handled logic directly
- Moved shared 'models' and 'services' directories from `web-vault` into `/libs/vault`

## 🚩 Constraints

The `VaultFilterService` reloads collections and folders when a new organization is set using `setOrganizationFilter()`, which aligns with the web experience. With this new design for desktop, we want to preserve all collections for organizations instead of filtered collections. Otherwise, all organizations are re-rendered as 'items' instead of 'groups' except for the selected org. The drawback is Folders aren't reloaded when a new organization is selected.   

## 📸 Screenshots


https://github.com/user-attachments/assets/10af81bd-6322-4e1d-be6f-733a3fd56f13

Hinton commented:
Please run prettier it should remove most teams from the review list due to the files being wrongly re-formatted ...

Hinton commented on apps/desktop/src/vault/app/vault-v3/vault-filter/filters/collection-filter.component.ts:
```diff
@@ -0,0 +1,40 @@
+  selector: "app-collection-filter",
+  templateUrl: "collection-filter.component.html",
+  standalone: true,
+  imports: [...VAULT_FILTER_IMPORTS],
```
**suggestion:**  Use explicit imports.

Hinton reviewed (commented):
Some initial reviews, I'll take second more in-depth look once they've been resolved.

lxiong-livefront commented on apps/desktop/src/app/app-routing.module.ts:
```diff
@@ -353,6 +361,18 @@ const routes: Routes = [
+        provide: PremiumUpgradePromptService,
+        useClass: DesktopPremiumUpgradePromptService,
+      },
+    ],
```
Addressed in [9383a78](https://github.com/bitwarden/clients/pull/17919/commits/9383a788ed0c1e9025220f7ecdbdb797d5260e8e)

lxiong-livefront commented on apps/desktop/src/vault/app/vault-v3/vault-filter/filters/collection-filter.component.ts:
```diff
@@ -0,0 +1,40 @@
+@Component({
+  selector: "app-collection-filter",
+  templateUrl: "collection-filter.component.html",
+  standalone: true,
```
Addressed in [d058592](https://github.com/bitwarden/clients/pull/17919/commits/d05859242e9b0c01a42265df79211c80bc8af91b)

lxiong-livefront commented on apps/desktop/src/vault/app/vault-v3/vault-filter/shared-filter-imports.ts:
```diff
@@ -0,0 +1,9 @@
+/**
+ * Common imports shared across all vault filter components.
+ */
+export const VAULT_FILTER_IMPORTS = [CommonModule, JslibModule, NavigationModule] as const;
```
Vault filter components all use `[appA11yTitle]`, which requires `JslibModule` (I kept this from their original base components). Should we remove these properties? Or is there another module you'd prefer to import for this?

Hinton replied:
That should be a standalone component that you can import from `libs/components`.

lxiong-livefront replied:
Updated in [f2ba5e4](https://github.com/bitwarden/clients/pull/17919/commits/f2ba5e4a38ffe27f63af6f19f941210e1ef01c35)

Hinton commented on apps/desktop/src/vault/app/vault-v3/vault-filter/vault-filter.component.ts:
```diff
@@ -0,0 +1,142 @@
+  protected readonly hideCollections = input(false);
+  protected readonly hideFavorites = input(false);
+  protected readonly hideTrash = input(false);
+  protected readonly hideOrganizations = input(false);
```
**suggestion:** We can probably simplify these and remove them since it's not customizeble.

lxiong-livefront commented:
> Some behavioral things noted while testing:
> 
> * Navigating to any filter should navigate you to the vault page.
> * It should be possible to select item types and folders while an Organization is selected.
> 
> You also have errors in other clients due to the file moves. (I suspect the path are absolute rather than using the `@bitwarden/vault` library.)

All vault filters should now correctly navigate to the vault page, and we are able to select types and folders while an org/collection is selected (addressed in [9f5dbb5](https://github.com/bitwarden/clients/pull/17919/commits/9f5dbb5e43db7a78b1b771328ca5d5d0b8b4b082)). I figured creating a 'DesktopRoutedVaultFilterBridge' would be the cleanest way to implement desktop specific navigation, but can handle routing in each filter component if that's preferred.

Also moved 'Collections' outside of 'Organizations', aligning with the web experience. Collections and Folders should now both be properly updated when an organization is selected. [383cb06](https://github.com/bitwarden/clients/pull/17919/commits/383cb0663c027b5502fed2e2917717d2dcfb0dfa)

Fixed broken imports in [383cb06](https://github.com/bitwarden/clients/pull/17919/commits/383cb0663c027b5502fed2e2917717d2dcfb0dfa)

eliykat reviewed (approved):
AC Team file changes are straightforward.

Hinton commented:
- The web builds are failing.
- Is there a way to reset "My vault" and "org" filters?

lxiong-livefront commented:
> FYI you have some type errors in the test files, you can run `npm run test:types` from the root of the repository to reproduce these locally.

Fixed type errors in test files in [2367ccb](https://github.com/bitwarden/clients/pull/17919/commits/2367ccbac8e8f6625532e0c8f97d731356a8af3e)

Also added an 'All vaults' and moved type filters under 'All items' in [d433fa2](https://github.com/bitwarden/clients/pull/17919/commits/d433fa22d58bdc87092423d88c9ff48d1e889ff5). I tested and found some strange behavior with 'My vault' items disappearing on page reload, but fixed that in this commit in `vault.component`. Let me know if you notice any other issues while testing!

Hinton reviewed (commented):
We're seeing lots of failures in building web and running tests related to bad imports.  

For tests run `npm run test` in the root of the project.
For building web run `build:bit:dev:watch` in `apps/web`.

Hinton commented on apps/desktop/src/app/services/desktop-routed-vault-filter-bridge.service.ts:
```diff
@@ -0,0 +1,38 @@
+    const extras = this.desktopRoutedVaultFilterService.createRoute(filter)[1];
+    const vaultCommands = [DesktopRoutedVaultFilterBridgeService.VAULT_ROUTE];
+
+    // FIXME: Verify that this floating promise is intentional. If it is, add an explanatory comment and ensure there is proper error handling.
```
```suggestion
```

Hinton commented on apps/desktop/src/app/services/desktop-routed-vault-filter-bridge.service.ts:
```diff
@@ -0,0 +1,38 @@
+    super(router, routedVaultFilterService, vaultFilterService);
+    this.desktopRouter = router;
+    this.desktopRoutedVaultFilterService = routedVaultFilterService;
+  }
```
**question:** Why are we creating duplicative variables for these services? Can you just update the visibility from private to protected in the parent service?

Hinton commented on apps/desktop/src/vault/app/vault-v3/vault.component.ts:
I don't believe there is a container ref for this and it's unused. Can we remove this?

Hinton reviewed (commented):
I think once these are addressed we can mark this as ready for review to get vault and AC reviews.

lxiong-livefront commented on apps/desktop/src/app/services/desktop-routed-vault-filter-bridge.service.ts:
I addressed this round of feedback in [26b2c5a](https://github.com/bitwarden/clients/pull/17919/commits/26b2c5a03e3837b583f46ca594029bd877d33ca7)

I decided to remove `DesktopRoutedVaultFilterBridge` and made base route an injectable per your suggestion. Let me know if this was what you had in mind - happy to go about it differently!

eliykat commented on libs/common/src/admin-console/utils/collection-utils.spec.ts:
```diff
@@ -1,3 +1,4 @@
+// eslint-disable-next-line no-restricted-imports
 import { CollectionView } from "@bitwarden/admin-console/common";
```
We shouldn't disable this rule - it prevents circular imports. I suggest the simplest solution for now is to move the other files you need into `libs/common/src/admin-console` alongside this one. `@bitwarden/admin-console/common` is an old pattern that will go away anyway.

If this cascades to moving/updating too many files, I can check how hard and fast this rule is and potentially make it a follow up ticket instead.

Hinton replied:
@eliykat I think the files should be moved now.

eliykat replied:
lgtm
